### PR TITLE
Enable Qdrant vectorstore

### PR DESCRIPTION
**Pull request by yuwenzho**
## Type of Change

feature 
API changed or not: no

## Description

Enable Qdrant vectorstore

## How has this PR been tested?
 
CI

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py [hunk 1]
```diff
@@ -134,9 +134,11 @@ def __init__(self,
 
         if self.vector_database == "Chroma":
             self.database = Chroma()
+        elif self.vector_database == "Qdrant":
+            self.database = Qdrant
```
Comment (XuhuiRen) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py:138]: add a "()", should keep the same format as Chroma

File: intel_extension_for_transformers/langchain/vectorstores/qdrant.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+            collection_name (Optional[str], optional): Name of the Qdrant collection to be used. 
+                Defaults to _LANGCHAIN_DEFAULT_COLLECTION_NAME.
+            force_recreate (bool, optional): _description_. Defaults to False.
+        """
+        if sum([param is not None for param in (location, url, host, persist_directory)]) == 0:
+            # One of 'location', 'url', 'host' or 'persist_directory' should be specified.
+            persist_directory = _DEFAULT_PERSIST_DIR
+            if sign == "child":
+                persist_directory = persist_directory + "_child"
+        assert not (sign == "child" and not persist_directory), \
```
Comment (XuhuiRen) [inline intel_extension_for_transformers/langchain/vectorstores/qdrant.py:86]: only check "persist_directory" is ok here. "sign" is an optional setting for child-parent retriever

File: intel_extension_for_transformers/langchain/vectorstores/qdrant.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                        Use data stored in another collection to initialize this collection.
+                    on_disk (Optional[bool], optional): if True, vectors will be stored on disk.
+                        If None, default value will be used.
+        """
+        if sum([param is not None for param in (location, url, host, persist_directory)]) == 0:
+            # One of 'location', 'url', 'host' or 'persist_directory' should be specified.
+            persist_directory = _DEFAULT_PERSIST_DIR
+            if sign == "child":
+                persist_directory = persist_directory + "_child"
+        assert not (sign == "child" and not persist_directory), \
```
Comment (XuhuiRen) [inline intel_extension_for_transformers/langchain/vectorstores/qdrant.py:195]: same to the above issue

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py (hunk 1 above)
Comment (yuwenzho) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py:138]: Qdrant init required 2 positional arguments: 'client' and 'collection_name'. Only @classmethod functions will be called below, so maybe we can avoid initializing the vectorstore classes here?

**Review by XuhuiRen — APPROVED** [review]

**Comment by XuhuiRen** [comment]
pylint error is caused by the code refactorization of langchain in December. Block the langchain version at first with https://github.com/intel/intel-extension-for-transformers/pull/1118.
Will root cause the changes in the new langchain version and change the import path.

**Comment by hshen14** [comment]
> pylint error is caused by the code refactorization of langchain in December. Block the langchain version at first with #1118. Will root cause the changes in the new langchain version and change the import path.

Pls add a fixing in NeuralChat first to make this pr move forward, while keeping eyes on new langchain version.

**Pull request merged**